dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -140,21 +174,71 @@ export const Picture = ({
 	isMainMedia = false,
 	isLazy = true,
 }: Props) => {
-	const hdpiSources = getSources(role, 'hdpi', imageSources);
-	const mdpiSources = getSources(role, 'mdpi', imageSources);
-	const fallbackSrc = getFallback(role, 'hdpi', imageSources);
-	const sizes = getSizes(role, isMainMedia, format);
+	const hdpiSourceSets = getSourcesForRoleAndResolution(
+		imageSources,
+		role,
+		'hdpi',
+	);
+	const mdpiSourceSets = getSourcesForRoleAndResolution(
+		imageSources,
+		role,
+		'mdpi',
+	);
+	const fallbackSrc = getFallback(
+		hdpiSourceSets.length ? hdpiSourceSets : mdpiSourceSets,
+	);
+
+	// Create an array of breakpoints going from highest to lowest, with 0 as the final option
+	const breakpointSizes: [number, number][] = [
+		...Object.values(breakpoints).reverse(),
```

Comment:
We could replace this with an explicit array with each breakpoint declared inside. More verbose but the breakpoints are static and it will be easier to read